File: AD-HOC/2423.js
let [F, O, L] = lines.shift().split(' ').map(Number);

let maxF = Math.floor(F/2);
let maxO = Math.floor(O/3);
let maxL = Math.floor(L/5);

let maximo = Math.min(maxF, maxO, maxL);

console.log(maximo);

File: AD-HOC/2423.py
import math

entrada = list(map(int, input().split()))
F = entrada[0]
O = entrada[1]
L = entrada[2]

maxF = math.floor(F/2)
maxO = math.floor(O/3)
maxL = math.floor(L/5)

maximo = min(maxF, maxO, maxL)

print(maximo)
